add_to_queue() at given position

ghcs27 opened the pull request:
This PR adds optional `position` and `as_next` arguments to `add_uri_to_queue()` and `add_to_queue()` to specify at which position the item should be inserted into the queue and whether this item should be played next.

KennethNielsen commented on soco/core.py:
```diff
@@ -1261,28 +1261,36 @@ def get_sonos_playlists(self, *args, **kwargs):
     @only_on_master
-    def add_uri_to_queue(self, uri):
+    def add_uri_to_queue(self, uri, position=0, as_next=False):
         """Adds the URI to the queue.
```
While it doesn't pertain to this PR at all, could I convince you to convert the docstring to Napoleon syntax and also to exchange the bulk of the two, so that the main (full) docstring is in add_to_queue and the docstring to add_uri_to_queue then refers to the other one.

KennethNielsen commented:
I can't seem to get the `as_next` argument to work (even in normal play mode). Both with and without a `position` argument, the `as_next` doesn't do anything for me. I did something like this, with an album already playing in the queue:

```python
import soco

for z in soco.discover():
    if z.player_name == "Stue":
        break

def print_queue(z):
    for item in z.get_queue():
        print(item.title)
    print()

print_queue(z)

t = z.music_library.get_tracks(search_term="Enter")[0]
print(t.title)
print()

z.add_to_queue(t, as_next=True)

print_queue(z)

```

ghcs27 commented:
I just did another test and realized `as_next` works only if shuffle is enabled.
It does not change the queue in any way. Rather, in shuffle mode the player selects the next title in beforehand, and `as_next=True` sets the new item as the next track.
Sonos uses this feature for the 'Play Next' command.
I have updated the docstring to point out that the argument only works in shuffle mode, and converted it to Napoleon syntax.
Please have a look at the changes and check whether, with this clarification, `as_next` also works for you.

KennethNielsen commented on soco/core.py:
```diff
@@ -1283,7 +1275,15 @@ def add_uri_to_queue(self, uri, position=0, as_next=False):
-        For Arguments see `add_uri_to_queue`"""
+
+        Args:
+            queueable_item (str): The item to be added to the queue
```
Thanks for doing the doc string work as well. Just two detail, the argument type for `queueable_item` is no longer `str` after you moved it. It should now be `DidlObject or MusicServiceItem`.

KennethNielsen commented on soco/core.py:
```diff
@@ -1283,7 +1275,15 @@ def add_uri_to_queue(self, uri, position=0, as_next=False):
+        Args:
+            queueable_item (str): The item to be added to the queue
+            position (int): The index (1-based) at which the URI should be
+            added.
```
I think the "added." was somehow left hanging. It should either be moved to the line above if that doesn't make it too long or indent it one more level to work correctly as Napoleon.

KennethNielsen commented:
Thanks for clarification on the `as_next` argument. I think a doc string clarification for that will do just fine. I made a new review for a few more nit-picks concerning the doc string updates I coerced you into doing ;) Thanks.

ghcs27 commented:
No problem 😄

KennethNielsen commented:
Merged. Thanks for your contribution.

Can you please add a comment to the release notes thread #440?
